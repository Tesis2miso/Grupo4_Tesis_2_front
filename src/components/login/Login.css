.login {
    background-color: rgb(26, 53, 235);
    min-height: 100vh;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    font-size: calc(10px + 2vmin);
    color: white;
  }


.label_error
{
  font-size:small;
  color: red;
}

.goToSignup {
  background-color: transparent;
  border: none;
  margin-bottom: 20px;
  font-size: 15px;
  color: lightgray;
}

a {
  font-size: 20px;
  color: rgb(226, 189, 247);
}
a:visited {
  color: rgb(226, 189, 247);
}
a:hover, a:active, a:focus {
  color: rgb(156, 111, 111);
}

.ddlLanguage{
  background-color: antiquewhite;
  float: right;
  margin-right: 10px;
}
