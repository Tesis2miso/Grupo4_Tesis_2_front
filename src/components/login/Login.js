import { useState, Fragment } from 'react';
import axios from "axios";
import './Login.css';
import { useEffect } from 'react';
import { useNavigate } from 'react-router-dom';
import { useTranslation } from "react-i18next";

function Login(props) {
    const [loginForm, setloginForm] = useState({
        email: "",
        password: ""
    })
    const [errorCred, setErrorCred] = useState();
    const [errorCaps, setErrorCaps] = useState();
    const [isCapsLockOn, setIsCapsLockOn] = useState(false);
    const navigate = useNavigate()
    const { loggedIn } = props
    const { t } = useTranslation();
    useEffect(() => {
        if (loggedIn) {
            navigate('/')
        }
    })

    const checkCapsLock = (event) => {
        if (event.getModifierState('CapsLock')) {
            setIsCapsLockOn(true);
        } else {
            setIsCapsLockOn(false);
        }
    };

    function logMeIn(event) {
        event.preventDefault()
        axios({
            method: "POST",
            url: `${process.env.REACT_APP_BASE_PATH}/specialist/login`,
            data: {
                email: loginForm.email,
                password: loginForm.password
            }
        })
            .then((response) => {
                props.setToken(response.data.access_token);
                localStorage.setItem("userName", response.data.username.username)
                navigate('/')
            }).catch((error) => {
                setloginForm(({
                    email: "",
                    password: ""
                }))
                setErrorCred('Invalid Username or Password')
            })
    }

    function handleChange(event) {
        const { value, name } = event.target
        setloginForm(prevNote => ({
            ...prevNote, [name]: value
        })
        )
    }

    const goToSignup = (event) => {
        event.preventDefault()
        navigate('/signup')
    }

    return (
        <Fragment>
            <form className="login">
                <h1>DermoApp</h1>
                <br></br>
                <input id="username" onChange={handleChange}
                    type="email"
                    text={loginForm.email}
                    name="email"
                    placeholder={t("email")}
                    value={loginForm.email} />
                <input id="password" onChange={handleChange}
                    data-testid="password"
                    type="password"
                    text={loginForm.password}
                    name="password"
                    placeholder={t("password")}
                    value={loginForm.password} />
                <div><label className='label_error'>{errorCred}</label></div>
                {isCapsLockOn && (
                    <p className='label_error'>{t('errorBMayus')}</p>
                )}                
                <button id="goToSignup" className="goToSignup" onClick={goToSignup}>{t('signin')}</button>
                <button id="submitbtn1" onClick={logMeIn}>{t('submit')}</button>
                <a href='#'>{t("forgotpwd")}</a>
            </form>
        </Fragment>
    );
}

export default Login;