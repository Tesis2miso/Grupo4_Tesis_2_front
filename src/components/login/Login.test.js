import '@testing-library/jest-dom';
import { mount, shallow } from 'enzyme';
import Login from './Login';
import axios from "axios";
import { render } from '@testing-library/react';
import { act } from 'react-dom/test-utils';

const mockedUsedNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockedUsedNavigate,
}));
jest.mock('axios', () => jest.fn());
jest.mock('react-i18next', () => ({
  useTranslation: () => {
    return {
      t: (str) => str,
      i18n: {
        changeLanguage: () => new Promise(() => { }),
      },
    };
  },
  initReactI18next: {
    type: '3rdParty',
    init: () => { },
  }
}));

describe('Tests_Login', () => {
  let wrapper = null;
  var non_logged_in_wrapper = null;
  let credentials = null;


  beforeEach(() => {
    wrapper = shallow(<Login loggedIn={true} setToken={() => { }} />);
    non_logged_in_wrapper = shallow(<Login loggedIn={false} setToken={() => { }} />);
    credentials = {
      "access_token": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJmcmVzaCI6ZmFsc2UsImlhdCI6MTY3ODU5ODM0OSwianRpIjoiNDBjZTBmMmMtMWIwYy00NDg4LThkZjktOTNjZTVlNTQ4MWYyIiwidHlwZSI6ImFjY2VzcyIsInN1YiI6ImUucXVpbnRlcm9wQHVuaWFuZGVzLmVkdS5jbyIsIm5iZiI6MTY3ODU5ODM0OSwiZXhwIjoxNjc4NTk5MjQ5fQ.tVe4Y1KnZQMbamBsiKXrBYhW-frLH4xfjMv9OnnPjIw",
      "message": "Access granted",
      "username": {
        "id": 1,
        "last_name": "Quintero Pinto",
        "name": "Emilson D",
        "username": "quinteroe"
      }
    }
  });

  test('show_Login', () => {
    render(<Login loggedIn={true} setToken={() => { }} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('login if logged in', () => {
    expect(wrapper.find('h1').text()).toEqual('DermoApp');
    expect(wrapper.find('#username')).toHaveLength(1);
    expect(wrapper.find('#password')).toHaveLength(1);
    expect(wrapper.find('#submitbtn1').text()).toEqual('submit');
  });

  test('login if not logged in', () => {
    expect(non_logged_in_wrapper.find('h1').text()).toEqual('DermoApp');
    expect(non_logged_in_wrapper.find('#username')).toHaveLength(1);
    expect(non_logged_in_wrapper.find('#password')).toHaveLength(1);
    expect(non_logged_in_wrapper.find('#submitbtn1').text()).toEqual('submit');
  });

  test('input valid credentials', () => {
    axios.mockResolvedValueOnce(Promise.resolve({ response: { data: { access_token: '' } } }));
    wrapper.find('#username').simulate('change', { target: { value: 'test@gmail.com', name: 'username' } })
    wrapper.find('#password').simulate('change', { target: { value: '123456', name: 'password' } })
    wrapper.find('#submitbtn1').simulate('click', {
      preventDefault: () => { }
    });
  });

  test('input invalid credentials', () => {
    axios.mockResolvedValueOnce(Promise.reject({ response: { data: { mssg: '' } } }));
    wrapper.find('#username').simulate('change', { target: { value: 'test@gmail.com', name: 'username' } })
    wrapper.find('#password').simulate('change', { target: { value: '123456', name: 'password' } })
    wrapper.find('#submitbtn1').simulate('click', {
      preventDefault: () => { }
    });
  });

  test('input mayus credentials', () => {
    const event = new KeyboardEvent('keydown', { 'keyCode': 20 });
    document.dispatchEvent(event);
    expect(wrapper.find('#label_error')).toHaveLength(0);
  });

  test('load consults', async () => {
    axios.mockResolvedValueOnce(Promise.resolve({ data: credentials }));
    const prop = {
      OnValChange: jest.fn()
    }
    await act(async () => {
      mount(<Login {...prop} />)
    });
  })

  test('go to signup', () => {
    wrapper.find('#goToSignup').simulate('click', {
      preventDefault: () => { }
    });
  })
})