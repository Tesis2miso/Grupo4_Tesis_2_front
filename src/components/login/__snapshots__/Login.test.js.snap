// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests_Login show_Login 1`] = `
<Fragment>
  <form
    className="login"
  >
    <h1>
      DermoApp
    </h1>
    <br />
    <input
      id="username"
      name="email"
      onChange={[Function]}
      placeholder="email"
      text=""
      type="email"
      value=""
    />
    <input
      data-testid="password"
      id="password"
      name="password"
      onChange={[Function]}
      placeholder="password"
      text=""
      type="password"
      value=""
    />
    <div>
      <label
        className="label_error"
      />
    </div>
    <button
      className="goToSignup"
      id="goToSignup"
      onClick={[Function]}
    >
      signin
    </button>
    <button
      id="submitbtn1"
      onClick={[Function]}
    >
      submit
    </button>
    <a
      href="#"
    >
      forgotpwd
    </a>
  </form>
</Fragment>
`;
